### PR TITLE
On demand

### PR DESCRIPTION
**Pull request by amiturgman**
@morsh please review

### CODE REVIEW

**Comment by amiturgman** [comment]
@morsh - please review. let me know if you have no comments/questions so I can merge.

File: examples/handlers/set_alarm.js [hunk 1]
```diff
@@ -1,15 +1,17 @@
...
-  var intent = session.dialogData[data.source];
+  var intent = session.privateConversationData[data.source];
   var alarmTime = null;
   if (intent.actions[0].parameters[0].name == "time") {
-
-    alarmTime = intent.actions[0].parameters[0].value[0].resolution.time;
+    // alarmTime = intent.entities...
+    // use intent.entities to extract relevant information                
+    // assuming extracted alarmTime                                       
+    alarmTime = '2016-10-10 10:10';
```
Comment (morsh) [inline examples/handlers/set_alarm.js:10]: You can add the old code as a comment to show a sample of how to get the resolved time

File: src/GraphDialog.ts [hunk 2]
```diff
@@ -81,6 +95,40 @@ export interface IGraphDialog {
...
+  /**
+   * Reloads scenarios for this instance.
+   * Use this when the scenarios were updated on the remote data source.
+   * After calling this method you'll probably want to call the restartDialog API to restart the updated dialog.
+   * 
+   * @returns {Promise<IGraphDialog>}
+   * 
+   * @memberOf IGraphDialog
+   */
+  reload(): Promise<IGraphDialog>;
```
Comment (morsh) [inline src/GraphDialog.ts:131]: great documenting!

**Review by morsh — COMMENTED** [review]
I haven't tried running those changes (or the ones in bot-trees). All in all the changes look good and well documented.

I don't know if those changes require an update to the README file, but we should probably start writing more README files to make it easier to follow the changes.

**Pull request merged**